sidekick_core/lib/src/commands/format_command.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:glob/glob.dart';
+import 'package:meta/meta.dart';
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:yaml/yaml.dart';
+
+/// Formats the Code for all Flutter/Dart packages in the repository
+class FormatCommand extends Command {
+  @override
+  String get description => 'Formatting the code for all packages';
+
+  @override
+  String get name => 'format';
+
+  /// glob patterns of packages whose files should not be formatted.
+  ///
+  /// Search starts at repository root.
+  ///
+  /// Example project layout:
+  ///
+  /// ```
+  /// project-root
+  /// ├── packages
+  /// │   ├── package1
+  /// │   ├── package2
+  /// │   └── circle
+  /// └── third_party
+  ///     ├── circle
+  ///     │   ├── packageA
+  ///     │   └── packageB
+  ///     └── square
+  /// ```
+  ///
+  /// - Use `packages/package1/**` to exclude only `packages/package1`.
+  /// - Use `**/circle/**` to exclude `packages/circle` as well as
+  ///   `third_party/circle/packageA` and `third_party/circle/packageB`.
+  final List<String> excludeGlob;
+
+  FormatCommand({
+    this.excludeGlob = const [],
+  }) {
+    argParser.addOption(
+      'package',
+      abbr: 'p',
+    );
```

Comment:
That's fine. In the future I want to convert our commands to use `DartPackage.fromArgResults()`, meaning the package comes from the `argsResult!.rest` which is the path to the package.

It's good that you kept consistency with the DepsCommand